### ISSUE TITLE
NuGet is deprecating ODATA filters on V2 endpoints. We need to migrate our use of their endpoints

### ISSUE THREAD

**Issue by brmenchl** [GlitchEnzo/NuGetForUnity#353]
## Description
It looks like NuGet is deprecating ODATA filters on V2 endpoints. We should migrate our usage of NuGet server endpoints to support the change.

The NuGet "brown-out" window today has made it so when clicking "Install" on any package, a console error pops up saying the NuGet server responded with a 400 error.

Reference: https://github.com/NuGet/Announcements/issues/48

**Comment by bricelam** [comment]
Consider leveraging [NuGet.Protocol](https://www.nuget.org/packages/NuGet.Protocol).

**Comment by NorbertNemec** [comment]
I was just hit by this issue. Is there a workaround?

**Comment by NorbertNemec** [comment]
Sorry - reading the original post I realized the obvious workaround: wait for a few hours...

**Comment by FreakTheMighty** [comment]
This brown out hit us too 👊 🤯 😢 . To make things more confusing, things were working fine locally, perhaps where things were cached, but started to fail on our integration server.

**Comment by jwittner** [comment]
I've started digging into options on this. Tried getting the Nuget.* assemblies to work, but I end up having to bring in so many due to the dependency chain that I think it'd be problematic with conflicts in the long run.

Looking through NuGet server APIs and client source and I'm working on updating our API calls to support V3. I'm not planning to detect and continue support for V2 APIs, but move wholesale to only support V3. Curious if that breaks anyone's use cases.
